src/main/java/com/uber/cadence/internal/worker/autoscaler/ResizableSemaphore.java

```diff
@@ -0,0 +1,28 @@
+/*
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.internal.worker.autoscaler;
+
+import java.util.concurrent.Semaphore;
+
+public class ResizableSemaphore extends Semaphore {
```

Comment:
What functionality this class adds? Could we just use base class directly?

Reply:
Subclass is needed because reducePermits is protected method.